server/src/config.rs

```diff
@@ -1,43 +1,41 @@
 // Copyright (c) ZeroC, Inc.
 
-use crate::Backend;
-use tower_lsp::lsp_types::{ConfigurationItem, DidChangeConfigurationParams, Url};
+use tower_lsp::{
+    lsp_types::{ConfigurationItem, DidChangeConfigurationParams, Url},
+    Client,
+};
 
 #[derive(Default, Debug)]
 pub struct SliceConfig {
-    pub reference_urls: Option<Vec<Url>>,
+    pub references: Option<Vec<String>>,
+    pub root_uri: Option<Url>,
 }
 
 impl SliceConfig {
-    // Attempt to create a SliceConfig from the backend with the provided uri root.
-    pub async fn try_from_backend(
-        backend: &Backend,
-        root_uri: &Url,
-    ) -> tower_lsp::jsonrpc::Result<SliceConfig> {
-        let reference_directories = Self::fetch_reference_directories(backend).await?;
-        Self::create_config_from_directories(reference_directories, root_uri)
-    }
-
-    // Creates a SliceConfig from the provided configuration parameters and uri root.
-    pub fn from_configuration_parameters(
-        params: DidChangeConfigurationParams,
-        root_uri: &Url,
-    ) -> Self {
-        let reference_directories = Self::parse_reference_directories(&params);
-        Self::create_config_from_directories(reference_directories, root_uri).unwrap_or_default()
+    // Attempt to update the SliceConfig using the backend with the provided uri root.
+    pub async fn try_update(
```

Comment:
I would split into two functions. 
```
try_update_from_client
try_update_from_params
```

I don't see where you ever call  `try_udpate` where the wanted behavior is unknown.